python/lsst/pipe/tasks/setPrimaryFlags.py

```diff
@@ -69,47 +75,58 @@ def run(self, sources, skyMap, tractInfo, patchInfo, includeDeblend=True):
-            except Exception:
-                self.log.warn("merge_peak is not set for pseudo-filter %s" % filt)
+            pseudoFilterKeys = []
+            for filt in self.config.pseudoFilterList:
+                try:
+                    pseudoFilterKeys.append(self.schema.find("merge_peak_%s" % filt).getKey())
+                except Exception:
+                    self.log.warn("merge_peak is not set for pseudo-filter %s" % filt)
 
-        tractId = tractInfo.getId()
-        for source in sources:
-            centroidPos = source.getCentroid()
-            if numpy.any(numpy.isnan(centroidPos)):
-                continue
-            if source.getCentroidFlag():
-                # Use a slightly smaller box to guard against bad centroids (see above)
-                isPatchInner = shrunkInnerFloatBBox.contains(centroidPos)
-            else:
-                isPatchInner = innerFloatBBox.contains(centroidPos)
-            source.setFlag(self.isPatchInnerKey, isPatchInner)
+            tractId = tractInfo.getId()
+            for source in sources:
+                centroidPos = source.getCentroid()
+                if numpy.any(numpy.isnan(centroidPos)):
+                    continue
+                if source.getCentroidFlag():
+                    # Use a slightly smaller box to guard against bad centroids (see above)
+                    isPatchInner = shrunkInnerFloatBBox.contains(centroidPos)
+                else:
+                    isPatchInner = innerFloatBBox.contains(centroidPos)
+                source.setFlag(self.isPatchInnerKey, isPatchInner)
 
-            skyPos = source.getCoord()
-            sourceInnerTractId = skyMap.findTract(skyPos).getId()
-            isTractInner = sourceInnerTractId == tractId
-            source.setFlag(self.isTractInnerKey, isTractInner)
+                skyPos = source.getCoord()
+                sourceInnerTractId = skyMap.findTract(skyPos).getId()
+                isTractInner = sourceInnerTractId == tractId
+                source.setFlag(self.isTractInnerKey, isTractInner)
 
-            if nChildKey is None or source.get(nChildKey) == 0:
-                for pseudoFilterKey in pseudoFilterKeys:
-                    if source.get(pseudoFilterKey):
-                        isPseudo = True
-                        break
-                else:
-                    isPseudo = False
+                if nChildKey is None or source.get(nChildKey) == 0:
+                    for pseudoFilterKey in pseudoFilterKeys:
+                        if source.get(pseudoFilterKey):
+                            isPseudo = True
+                            break
+                    else:
+                        isPseudo = False
+
+                    source.setFlag(self.isPrimaryKey, isPatchInner and isTractInner and not isPseudo)
 
-                source.setFlag(self.isPrimaryKey, isPatchInner and isTractInner and not isPseudo)
+        else:
```

Comment:
Add comment indicating this is the single frame case, especially since the `if` was quite long

Reply:
Comments added prior to the start of both `if` and `else`.